### ISSUE TITLE
Weekly on-call version check - 11/22

### ISSUE THREAD

**Issue by oneilcin** [samsung-cnct/kraken-lib#981]
Steps to take:

- [ ] 1. Visit the kubernetes [release page](https://github.com/kubernetes/kubernetes/releases) and verify the latest minor version of the last 3 major releases (e.g. 1.6, 1.7 and 1.8) are supported by [kraken-lib](https://github.com/samsung-cnct/k2/blob/master/ansible/roles/kraken.config/files/config.yaml) and [kraken-tools](https://github.com/samsung-cnct/kraken-tools/blob/master/Dockerfile).  Update [kraken-lib config files](https://github.com/samsung-cnct/kraken-lib/tree/master/ansible/roles/kraken.config/files), [kraken-lib Jenkinsfile - line with conformance tests](https://github.com/samsung-cnct/kraken-lib/blob/master/Jenkinsfile#L13), and [kraken-tools](https://github.com/samsung-cnct/kraken-tools/blob/0e9362dafb0f4a679c8990e853a6f678546e1e2f/Dockerfile#L25). The kubernetes [CHANGELOG](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG.md) can also be useful when determining the latest patch version for an older minor version, and when determining if any changes are necessary beyond updating the default versions used by kraken-lib et al.
Kraken-tools: If updating any versions, checksum must be updated as well. To do so: Download binary via `wget -q [URL]` and then generate via `shasum -a 256 [binary]`.
- [ ] 2. Visit the helm [release page](https://github.com/kubernetes/helm/releases) and see if there is a new release. Since helm doesn't seem to have minor release trains like kubernetes does, the oldest supported k8s (e.g. 1.6 and 1.7) are essentially frozen (i.e. you only need to look for the latest). If necessary, update [kraken-tools](https://github.com/samsung-cnct/kraken-tools/blob/master/Dockerfile) with current version of helm ( as of k8s v1.8, this is the variable `K8S_VERSION_1_8`)
- [ ] 3. Search for new docker related CVEs [here](https://www.docker.com/docker-cve-database) and [here](https://cve.mitre.org/cgi-bin/cvekey.cgi?keyword=docker). If any are found, please open a ticket against the [issues repo](https://github.com/samsung-cnct/issues/issues). Be sure to open the issue using the same format as https://github.com/samsung-cnct/issues/issues/3.   This ensures we can track which CVEs have been reviewed or not and will be important if we decide to automate this on-call work.
- [ ] 4. Check to see if [Alpine linux](https://hub.docker.com/_/alpine/) has been updated.  Kraken-tools has CI that should be sufficient.
- [ ] 5. Visit [quay.io](https://quay.io/repository/samsung_cnct/kraken-tools?tab=tags) and ensure the latest kraken-tools images does not have any vulnerabilities by checking the security scan of :latest to see that it has passed. 
- [ ] 6. Check if GKE CI jobs are taking over 5 minutes, https://common-jenkins.kubeme.io/job/K2/ 
 If so, adjust default Kubernetes version to match GKE default Kubernetes version to get CI runs back to 3-4 minutes. Use `gcloud container get-server-config --zone=us-west1-a --format json` to interrogate GKE for the current default version of K8s.
- [ ] 7. Kubernetes Conformance Dependencies - Check when we are moving to a new minor version of Kubernetes
   - etcd
   - kube-dns
   - docker (blocked by https://github.com/samsung-cnct/kraken-lib/issues/946) but once that has been closed, we can pin Docker
Check if these changed when we are moving to a new minor version of Kubernetes.  Tips for looking. https://k8s-testgrid.appspot.com/
Select 'sig-scalability' -> 'sig-scalability-gce' -> 'gce-1.9' ->About (see these results in Gubernater) -> choose a green build -> Artifacts -> Artifacts folder -> scroll down to nodes.yaml
- [ ] 8. Non-Kubernetes Dependencies
- [coreOS](https://coreos.com/releases/)
- [Calico - node and cni component] (https://docs.projectcalico.org/v2.6/releases/)
- [Weave](https://github.com/weaveworks/weave/releases)
- [Flannel](https://github.com/coreos/flannel/releases)
Note: You only need to change the AWS config for fabric changes.  Weave is not CI tested by default.  If changing Weave version, you need to run a test with an nginx pod.  Make sure you can curl to it from a node other than the node the nginx pod is running on.
- [ ] 9. Create a [new issue](https://github.com/samsung-cnct/issues/issues/new) for next week. Be sure to:
   - include this text in the description
   - assign it to next person in the list below
   - label with `operations` and `priority-p1`
   - move this issue from the **Triage** list to the  **ToDo** column in the [commons project](https://github.com/orgs/samsung-cnct/projects/1#card-4154822)
- [ ] 10. When you have done all the steps above, write up a comment like [this](https://github.com/samsung-cnct/issues/issues/45#issuecomment-327950490) that states:
   - the current versions of kubernetes and helm, and whether or not you updated them. 
   - the date that you checked the CVEs and whether or not you created a ticket to deal with it
   - link to the vulnerability scan results for the current version of the `:latest` tag
   - a link to the PR to change the default GKE version if done
   - a link to the issue you created for next week, assigned to the next person on rotation
   - move this issue to the **Review** column in the [commons project](https://github.com/orgs/samsung-cnct/projects/1#card-4154822)

On-call rotation order:

@davidewatson, @leahnp, @DStorck, @yenicapotediaz, @coffeepac, @oneilcin, @guineveresaenger , @joejulian  

Last review was: #90

**Comment by oneilcin** [comment]
Opened under issues
